packages/flutter/lib/src/foundation/assertions.dart

```diff
@@ -20,6 +20,96 @@ typedef DiagnosticPropertiesTransformer = Iterable<DiagnosticsNode> Function(Ite
 /// and other callbacks that collect information describing an error.
 typedef InformationCollector = Iterable<DiagnosticsNode> Function();
 
+/// Partial information from a stack frame for stack filtering purposes.
+///
+/// See also:
+///
+///  * [SubStackFilter], which uses this class to compare against [StackFrame]s.
+@immutable
+class PartialStackFrame {
+  /// Creates a new [PartialStackFrame] instance. All arguments are required and
+  /// must not be null.
+  const PartialStackFrame({
+    @required this.className,
+    @required this.method,
+    @required this.packagePath,
+  }) : assert(className != null),
+       assert(method != null),
+       assert(packagePath != null);
+
+  /// The class name for the method.
+  ///
+  /// On web, this is ignored, since class names are not available. Top level
+  /// methods should use the empty string.
+  final String className;
+
+  /// The method name for this frame line.
+  ///
+  /// On web, private methods are wrapped with `[]`.
+  final String method;
+
+  /// The relative path to the file for this method and class.
+  ///
+  /// This is required for disambiguation.
+  final String packagePath;
+
+  @override
+  String toString() {
+    if (kIsWeb) {
+      final String webMethod = method.startsWith('_') ? '[$method]' : method;
+      return '$packagePath: $webMethod';
+    }
+    return '$packagePath: $className.$method';
+  }
+}
+
+/// A class that matches sublists of stack frames for additional filtering
+/// of [FlutterError.defaultStackFilter].
+///
+/// This class provides a list of partial stack [frames] to match against and a
+/// replacement string for matches.
+///
+/// See also:
+///
+///   * [FlutterError.addDefaultStackFilter], a method to register additional
+///     stack filters for [FlutterError.defaultStackFilter].
+///   * [StackFrame], a class that can help with parsing stack frames.
+///   * [PartialStackFrame], a class that helps match partial method information
+///     to a stack frame.
+class SubStackFilter {
```

Comment:
would it make sense to have a mechanism whereby you can match frames more elaborately, e.g. by implementing your own matchFrames?

Reply:
I wouldn't want my subclass to have to carry around the `frames` and `numFrames` and `replacement` members that are all just part of this one's implementation, though...

Reply:
Ahh. I can fix that.

Reply:
See latest commit.

Although it no longer tells you how many elements were elided - I could probably make that still work but it seems more complicated than its worth at this point.